kafka/consumer_test.go

```diff
@@ -429,3 +430,223 @@ func TestConsumerLog(t *testing.T) {
+			c, c.GetRebalanceProtocol(), len(ev.Partitions),
+			ev.Partitions)
+		// The application may update the start .Offset of each
+		// assigned partition and then call IncrementalAssign().
+		// Even though this example does not alter the offsets we
+		// provide the call to IncrementalAssign() as an example.
+		err := c.Assign(ev.Partitions)
+		if err != nil {
+			panic(err)
+		}
+
+	case RevokedPartitions:
+		atomic.AddInt32(&revokedEvents, 1)
+		fmt.Fprintf(os.Stderr,
+			"%s rebalance: %d partition(s) revoked: %v\n",
+			c.GetRebalanceProtocol(), len(ev.Partitions),
+			ev.Partitions)
+		if c.AssignmentLost() {
+			// Our consumer has been kicked out of the group and the
+			// entire assignment is thus lost.
+			fmt.Fprintf(os.Stderr, "%% Current assignment lost!\n")
+		}
+
+		// The client automatically calls Unassign() unless
+		// the callback has already called that method.
+	}
+
+	return nil
+}
+
+func testPoll(c *Consumer, doneChan chan bool) {
+	run := true
+	for run {
+		select {
+		case <-doneChan:
+			run = false
+
+		default:
+			ev := c.Poll(100)
+			if ev == nil {
+				continue
+			}
+			switch e := ev.(type) {
+			case *Message:
+				fmt.Printf("Message on %s:\n%s\n",
+					e.TopicPartition, string(e.Value))
+				if e.Headers != nil {
+					fmt.Printf("Headers: %v\n", e.Headers)
+				}
+
+			case Error:
+				// Errors should generally be
+				// considered informational, the client
+				// will try to automatically recover.
+				fmt.Printf("Error: %v: %v for "+
+					"consumer %v\n", e.Code(), e, c)
+
+			default:
+				fmt.Printf("Ignored %v for consumer %v\n",
+					e, c)
+			}
+		}
+	}
+}
+
+// TestConsumerCloseForStaticMember verifies the rebalance
+// for static membership.
+// According to KIP-345, the consumer group will not trigger rebalance unless
+// 1) A new member joins
+// 2) A leader rejoins (possibly due to topic assignment change)
+// 3) An existing member offline time is over session timeout
+// 4) Broker receives a leave group request containing alistof
+//    `group.instance.id`s (details later)
+//
+// This test uses 3 consumers while each consumer joins after the assignment
+// finished for the previous consuemrs.
+// The expected behavior for these consuemrs are:
+// 1) First consumer joins, AssignedPartitions happens. Assign all the
+//    partitions to it.
+// 2) Second consumer joins, RevokedPartitions happens from the first consumer,
+//    then AssignedPartitions happens to both consumers.
+// 3) Third consumer joins, RevokedPartitions happens from the previous two
+//    consumers, then AssignedPartitions happens to all the three consumers.
+// 4) Close the second consumer, RevokedPartitions should not happen.
+// 5) Rejoin the second consumer, rebalance should not happen to all the other
+//    consumers since it's not the leader, AssignedPartitions only happened
+//    to this consumer to assign the partitions.
+//
+// The total number of AssignedPartitions is 7 and the total number of
```

Comment:
This should preferably be tracked per consumer so that we know that it was Consumer 2 that had the rebalance and not  the others.